File: sparkcql-core/src/main/scala/gov/va/sparkcql/translator/node/ContextDefNode.scala
package gov.va.sparkcql.translator.node

import org.hl7.elm.{r1 => elm}
import gov.va.sparkcql.translator._
import org.apache.spark.sql.Column
import org.apache.spark.sql.functions._

class ContextDefNode(val element: elm.ContextDef) extends Node {

  override def translate(env: Environment): Object = {
    // val relatedContextRetrieve = parent.descendants.collect { case n: SingletonFromNode => n }
    // val retrieves = relatedContextRetrieve.map(_.element.getOperand().asInstanceOf[elm.Retrieve].getDataType())
    
    // val contextDataFrame = element.getName() match {
    //   case "Patient" => env.dataAggregate.acquire(relatedContextRetrieve.getOperand().asInstanceOf[elm.Retrieve].getDataType())
    //   case "Practitioner" => throw new UnsupportedOperationException()
    //   case "Unfiltered" => throw new UnsupportedOperationException()
    //   case _ => throw new UnsupportedOperationException()
    // }

    // contextDataFrame
    ???
  }
}

File: transpiler/src/main/java/gov/va/transpiler/jinja/node/trackable/element/ContextDefNode.java
package gov.va.transpiler.jinja.node.trackable.element;

import org.hl7.elm.r1.ContextDef;

import gov.va.transpiler.jinja.printing.Segment;
import gov.va.transpiler.jinja.state.State;

public class ContextDefNode extends ElementNode<ContextDef> {

    public ContextDefNode(State state, ContextDef cqlEquivalent) {
        super(state, cqlEquivalent);
        state.setContext(getCqlEquivalent().getName());
    }

    @Override
    public Segment toSegment() {
        // Not printable
        throw new UnsupportedOperationException();
    }
}
